src/grelu/io/motifs.py

```diff
@@ -0,0 +1,158 @@
+"""
+Functions related to reading and writing MEME files
+"""
+
+from typing import Dict, List, Optional
+
+import numpy as np
+
+
+def read_meme_file(
+    file: str, names: Optional[List[str]] = None, n_motifs: Optional[int] = None
+) -> Dict[str, np.ndarray]:
+    """
+    Read a motif database in MEME format
+
+    Args:
+        file: The path to the MEME file
+        names: List of motif names to read
+        n_motifs: Number of motifs to read
+
+    Returns:
+        a dictionary in which the keys are motif names and the
+        values are the motif position probability matrices (PPMs)
+        as numpy arrays of shape (4, L).
+    """
+    from grelu.resources import get_meme_file_path
+
+    # Get file path
+    file = get_meme_file_path(file)
+
+    # Empty dictionary to store motifs read
+    motifs = dict()
+
+    # Number of motifs to read
+    if names is not None:
+        n_motifs = len(names)
+
+    # Read
+    with open(file, "r") as f:
+        name, width, i = None, None, 0
+
+        for line in f:
+            if name is None:
+                if line[:5] == "MOTIF":
+                    name = line.replace("MOTIF ", "").strip("\r\n")
+                else:
+                    continue
+
+            elif width is None:
+                if line[:6] == "letter":
+                    width = int(line.split()[5])
+                    ppm = np.zeros((width, 4))
+
+            elif i < width:
+                ppm[i] = list(map(float, line.strip("\r\n").split()))
+                i += 1
+
+            else:
+                # Add the new motif to motifs dict
+                if names is None:
+                    motifs[name] = ppm.T
+                elif name in names:
+                    motifs[name] = ppm.T
+
+                # If all the required motifs have been read, stop
+                if n_motifs is not None:
+                    if len(motifs) == n_motifs:
+                        break
+
+                # Otherwise, reset
+                name, width, i = None, None, 0
+
+    # Check how many motifs were found
+    print(f"Read {len(motifs)} motifs from file.")
+    if names is not None:
+        missing = set(names).difference(motifs.keys())
+        if len(missing) > 0:
+            print(f"{len(missing)} motifs were not found in the file: {missing}")
+
+    return motifs
+
+
+def read_modisco_report(
```

Comment:
this should ideally be a parser in the modiscolite repo itself, perhaps we should ask Jacob?

Reply:
We discussed it, the modiscolite repo is not being updated at present. So we should keep this here for now.